Tests/SonarScanner.MSBuild.Common.UnitTests/AnalysisConfig/AnalysisConfigExtensionsTests.cs

```diff
@@ -352,16 +352,33 @@ public void ConfigExt_GetAnalysisSettings_NoSettings()
             var config = new AnalysisConfig();
 
             // 1. No server settings
-            var provider = config.GetAnalysisSettings(false);
+            var provider = GetAnalysisSettingsIsolatedFromEnvironment(config, false);
             provider.Should().NotBeNull("Returned provider should not be null");
             provider.AssertExpectedPropertyCount(0);
 
             // 2. With server settings
-            provider = config.GetAnalysisSettings(true);
+            provider = GetAnalysisSettingsIsolatedFromEnvironment(config, true);
             provider.Should().NotBeNull("Returned provider should not be null");
             provider.AssertExpectedPropertyCount(0);
         }
 
         #endregion Tests
+
+        private static IAnalysisPropertyProvider GetAnalysisSettingsIsolatedFromEnvironment(AnalysisConfig config, bool includeServerSettings)
+        {
+            IAnalysisPropertyProvider provider = null;
+
+            // Make sure the test isn't affected by the hosting environment
+            // The SonarCloud VSTS extension sets additional properties in an environment variable that
+            // would affect the test
+            using (var scope = new EnvironmentVariableScope())
```

Comment:
The _EnvironmentVariableScope_ class was an existing test infrastructure class to enable reliable testing of components that use environment variables.
The scope takes a copy of the current environment variables when it is created and restores them when it is disposed. If a test wants to modify an environment variable it should call _scope.SetVariable(...)_.